src/daft-core/src/array/ops/arithmetic.rs

```diff
@@ -79,6 +81,50 @@ where
     }
 }
 
+impl Add for &Decimal128Array {
+    type Output = DaftResult<Decimal128Array>;
+    fn add(self, rhs: Self) -> Self::Output {
+        assert_eq!(self.data_type(), rhs.data_type());
```

Comment:
should we instead return an error instead of panicking if the types dont match?

Reply:
This should hopefully never happen since the series code should correctly match cast the types before this gets called.